dockerfiles/Dockerfile

```diff
@@ -0,0 +1,170 @@
+# syntax=docker/dockerfile:1.3-labs
+ARG OPENMS_VERSION=3.1.0
+ARG OPENMS_LIBRARY_BUILD_DIR="/openms-build"
+ARG OPENMS_LIBRARY_DIR="/OpenMS"
+ARG CONTRIB_TAG=latest
+ARG CONTRIB_BASE_IMAGE=ghcr.io/openms/contrib:${CONTRIB_TAG}
+ARG CONTRIB_BUILD_DIR="/contrib-build"
+ARG THIRDPARTY_BRANCH="master"
+ARG THIRDPARTY_DIR="/thirdparty"
+ARG NUM_BUILD_CORES=8
+ARG MAKEFLAGS="-j${NUM_BUILD_CORES}"
+
+
+### OpenMS base ###
+FROM ${CONTRIB_BASE_IMAGE} as base
+ARG OPENMS_VERSION
+
+# the main reason we're pulling this in is to make sure that the metadata
+# below is consistently formatted for releases
+LABEL version="${OPENMS_VERSION}"
+LABEL software.version="${OPENMS_VERSION}-Ubuntu22.04"
+LABEL org.opencontainers.image.source https://github.com/OpenMS/OpenMS
+
+
+### OpenMS library ###
+FROM base as library
+ARG CONTRIB_BUILD_DIR
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG MAKEFLAGS
+ENV MAKEFLAGS="${MAKEFLAGS}"
+
+# NOTE: this assumes that the context is root of the OpenMS repo
+COPY . ${OPENMS_LIBRARY_DIR}
+WORKDIR ${OPENMS_LIBRARY_BUILD_DIR}
+RUN /bin/bash -c "cmake -DCMAKE_BUILD_TYPE='Release' -DCMAKE_PREFIX_PATH='${CONTRIB_BUILD_DIR}/;/usr/;/usr/local' -DBOOST_USE_STATIC=OFF ${OPENMS_LIBRARY_DIR}"
+RUN make OpenMS
+LABEL software="OpenMS (library)"
+
+
+### OpenMS tools build ###
+FROM library as tools-build
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG MAKEFLAGS
+ARG THIRDPARTY_BRANCH
+ARG THIRDPARTY_DIR
+ENV MAKEFLAGS="${MAKEFLAGS}"
+
+RUN apt-get update && apt-get install -y --no-install-recommends --no-install-suggests openjdk-17-jdk
```

Comment:
JRE might be enough. Did you try?
It's for msgf and luciphor.

Reply:
I did not try, but good suggestion, thanks!

Reply:
Just to be clear, is this a build or runtime dependency?

Reply:
I think only runtime/testing with thirdparty.